Adapter/Common.php

```diff
@@ -95,6 +95,10 @@ public function cropResize($width = null, $height = null, $background = 'transpa
      */
     public function fixOrientation()
     {
+        if (!in_array(exif_imagetype($this->source->getInfos()), array(IMAGETYPE_JPEG, IMAGETYPE_TIFF_II, IMAGETYPE_TIFF_MM))) {
+            return $this;
```

Comment:
I'm against raising an exception because in my use case I get a list of image manipulation actions in a row, and then execute them all regardless of the file type (I don't know it yet when adding the actions), so returning `$this` and continuing execution for me is essential.